src/odemis/acq/stream/_base.py

```diff
@@ -586,6 +703,8 @@ def _prepare(self):
         logging.debug(u"Preparing stream %s ...", self.name.value)
         # actually indicate that preparation has been triggered, don't wait for
         # it to be completed
+        self._linkHwAxes()
+
```

Comment:
No. You shouldn't call it twice. Either call it in `prepare()`, in which case, ideally it'd be part of the future which is returned. Or you call it in `_is_active_setter()`. That second option is probably a little bit less nice, but easier, as you don't need to combine the futures.